### PR TITLE
FileDataLoader fails to read the file when size > INT32_MAX

### PR DESCRIPTION
**Pull request by cymbalrush**
On macOS, the `read` function will fail with an `EINVAL` error if the size parameter exceeds `INT32_MAX`. This update addresses the issue by adding a check to ensure that the read size does not surpass `INT32_MAX`. On Linux, the maximum permissible read size is 2,147,479,552 bytes ( < `INT32_MAX`), so attempting to read beyond this limit is inconsequential. 

Test Plan:

Exporting llama3 with `python -m examples.models.llama2.export_llama --checkpoint examples/models/llama-2-7B/consolidated.00.pth --params examples/models/llama-2-7B/params.json --coreml --disable_dynamic_shape -kv `  

Without fix
Fails with `invalid argument` error.

With fix
Succeeds.

### CODE REVIEW

File: extension/data_loader/file_data_loader.cpp [hunk 1]
```diff
@@ -189,7 +191,12 @@ Result<FreeableBuffer> FileDataLoader::load(
   size_t needed = size;
   uint8_t* buf = reinterpret_cast<uint8_t*>(aligned_buffer);
   while (needed > 0) {
-    ssize_t nread = ::read(fd_, buf, needed);
+    // read on macos would fail with EINVAL if size > INT32_MAX
```
Comment (dbort) [inline extension/data_loader/file_data_loader.cpp:194]: Please begin comments with uppercase letters and end with a period, to be consistent with the rest of the file. Thanks for adding this comment, though, to explain why this extra logic is necessary, and where to focus testing if it changes.

```suggestion
    // Reads on macOS will fail with EINVAL if size > INT32_MAX.
```

**Review by dbort — COMMENTED** [review]
Thank you so much for this fix! One super-minor request but the code changes look great.

I would suggest adding a unit test for this, but I really don't want our test jobs creating and reading a >2GB file. But because there's no automated testing, please add a `Test Plan:` section to the PR description explaining how you tested that this works.

**Comment by cymbalrush** [comment]
> Thank you so much for this fix! One super-minor request but the code changes look great.
> 
> I would suggest adding a unit test for this, but I really don't want our test jobs creating and reading a >2GB file. But because there's no automated testing, please add a `Test Plan:` section to the PR description explaining how you tested that this works.

That's the reason for not adding test, I didn't want to create a 2GB file for unit testing this.

File: extension/data_loader/file_data_loader.cpp (hunk 1 above)
Comment (dbort) [reply extension/data_loader/file_data_loader.cpp:194]: When you get a chance please update this comment, then we can land the PR

Comment (cymbalrush) [reply extension/data_loader/file_data_loader.cpp]: Done.

**Pull request closed without merging**